src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -150,56 +176,100 @@ loadMesh( Path const & filePath )
   return loadedMesh;
 }
 
+vtkNew< vtkCellArray > GetCellArray( vtkDataSet & mesh ) // replaces GetCells() that exist only in vtkUnstructuredGrid
+{
+  vtkNew< vtkCellArray > cells;
+  if( mesh.IsA( "vtkStructuredGrid" ) || mesh.IsA( "vtkUnstructuredGrid" ))
+  {
+    int numCell = mesh.GetNumberOfCells();
+    for( int c = 0; c < numCell; c++ )
+    {
+      cells->InsertNextCell( mesh.GetCell( c ));
+    }
+  }
+  else
+  {
+    double bounds[6];
+    mesh.GetBounds( bounds );
+    double xmin=bounds[0], xmax=bounds[1];
+    double ymin=bounds[2], ymax=bounds[3];
+    double zmin=bounds[4], zmax=bounds[5];
+    int dx = 1, dy = 1;
+    int nx = (xmax-xmin)/dx +1, ny = (ymax-ymin)/dy +1;
+    //int nz = (zmax-zmin)/dz +1; // not used
+
+    for( int k=zmin; k<zmax; k++ )
```

Comment:
This loop looks weird--why does the indexing begin with `xmin`, `ymin `and `zmin`? Aren't they the "geographical" bounds of your system, and not integers? Are they given in meters? Or are they just integers?